libraries/rush-lib/src/logic/RepoStateFile.ts

```diff
@@ -182,6 +196,21 @@ export class RepoStateFile {
       this._modified = true;
     }
 
+    if (rushConfiguration.packageManager === 'pnpm' && rushConfiguration.subspacesFeatureEnabled) {
+      const packageJsonInjectedDependenciesHash: string | undefined =
+        subspace.getPackageJsonInjectedDependenciesHash();
+
+      // packageJsonInjectedDependenciesHash is undefined, means there is no injected dependencies for that subspace
+      // so we don't need to track the hash value for that subspace
+      if (
+        packageJsonInjectedDependenciesHash &&
+        packageJsonInjectedDependenciesHash !== this._packageJsonInjectedDependenciesHash
```

Comment:
If `packageJsonInjectedDependenciesHash` is undefined and `this._packageJsonInjectedDependenciesHash` isn't (or vise-versa), doesn't that indicate that the repo state has been modified?

Reply:
In the `RepoStateFile.js`, If `packageJsonInjectedDependenciesHash` is undefined, then we don't need to write the hash value. The actual check logic is in the `WorkspaceInstallManager.ts`